### DISCUSSION TITLE
Using shortcuts without modifiers

### DISCUSSION THREAD

**Discussion by N3C2L** [madprops/grasshopper#169] [Ideas]
It would be very comfortable for my fingers if one could use a single keyboard key as a shortcut for a command without needing modifiers. For example, pressing the "a" key would immediately execute a previously assigned shortcut (without an assigned modifier). However, this is currently not possible because when I press "a", this letter is immediately used as a search in the filter bar. But how could the user use just the "a" key as a shortcut without needing an additional modifier like Ctrl or Alt? Two things might be necessary for this:
 
A new setting where the filter search field would not be triggered immediately with every single keystroke. If a user has activated this new setting, they would then need a new command that focuses the filter search field at the top. And it would be good, when the filter search field is focused with this new command, the user can type normally as the shortcuts without modifiers don't work here anymore, becaue they need the keystrokes for typing inside the filter (if it's possible to implement).

I would find this very useful because without having to press additional modifiers, there would be more flexibility, especially when one's left fingers are on the home row keys (a,s,d,f), which are normal for touch typing. Then one could operate the mouse with the right hand, and with the left hand, for example, only the keys a,s,d,f could be pressed, and commands would be executed in GH. This would be very practical, and I hope that you might also find such an additional option good for GH.

**Comment by N3C2L** [comment]
My bad, sorry.
It is already possible to use single keys as hotkeys without triggering the filter bar.
I should have tested it before, sorry.

**Comment by N3C2L** [comment]
But maybe a new cmd for focusing the search filter bar on top would be useful?
(For people like me which use single keystrokes for cmds and could then focusing/activate the search filter bar with a cmd for filtering the tabs with it)

**Comment by madprops** [comment]
>But maybe a new cmd for focusing the search filter bar on top would be useful?

Yeah good idea, I'm adding that.

Also adding Unfocus Filter, just in case.

**Reply by N3C2L** [reply]
Wonderful, thanks!

**Comment by N3C2L** [comment]
Just another thought: How could a user know that the filter bar has been focused successfully with the cmd? Maybe the filterbar could also have a filter effect when the focus was sucessfull? This would help me as a visual feedback indicator.

**Reply by madprops** [reply]
Alright, I'm going to give it an outline. And a setting.

**Reply by N3C2L** [reply]
Great, thanks!
